satpy/tests/reader_tests/test_nwcsaf_hrw_nc.py

```diff
@@ -0,0 +1,234 @@
+    ("longitude_increment", "<f4"),
+    ("air_temperature", "<f4"),
+    ("air_pressure", "<f4"),
+    ("air_pressure_error", "<f4"),
+    ("air_pressure_correction", "<f4"),
+    ("wind_speed", "<f4"),
+    ("wind_from_direction", "<f4"),
+    ("quality_index_with_forecast", "i1"),
+    ("quality_index_without_forecast", "i1"),
+    ("quality_index_iwwg_value", "i1"),
+    ("tracer_correlation_method", "i1"),
+    ("tracer_type", "i1"),
+    ("height_assignment_method", "i1"),
+    ("orographic_index", "i1"),
+    ("cloud_type", "i1"),
+    ("correlation", "i1")
+])
+
+WIND_DTYPES = [
+    ("wind_idx", np.dtype("uint32")),
+    ("previous_wind_idx", np.dtype("uint32")),
+    ("number_of_winds", np.dtype("uint8")),
+    ("correlation_test", np.dtype("uint8")),
+    ("quality_test", np.dtype("uint16")),
+    ("segment_x", np.dtype("uint32")),
+    ("segment_y", np.dtype("uint32")),
+    ("segment_x_pix", np.dtype("uint32")),
+    ("segment_y_pix", np.dtype("uint32")),
+    ("latitude", np.dtype("<f4")),
+    ("longitude", np.dtype("<f4")),
+    ("latitude_increment", np.dtype("<f4")),
+    ("longitude_increment", np.dtype("<f4")),
+    ("air_temperature", np.dtype("<f4")),
+    ("air_pressure", np.dtype("<f4")),
+    ("air_pressure_error", np.dtype("<f4")),
+    ("air_pressure_correction", np.dtype("<f4")),
+    ("air_pressure_nwp_at_best_fit_level", np.dtype("<f4")),
+    ("wind_speed", np.dtype("<f4")),
+    ("wind_from_direction", np.dtype("<f4")),
+    ("wind_speed_nwp_at_amv_level", np.dtype("<f4")),
+    ("wind_from_direction_nwp_at_amv_level", np.dtype("<f4")),
+    ("wind_speed_nwp_at_best_fit_level", np.dtype("<f4")),
+    ("wind_from_direction_nwp_at_best_fit_level", np.dtype("<f4")),
+    ("wind_speed_difference_nwp_at_amv_level", np.dtype("<f4")),
+    ("wind_from_direction_difference_nwp_at_amv_level", np.dtype("<f4")),
+    ("wind_speed_difference_nwp_at_best_fit_level", np.dtype("<f4")),
+    ("wind_from_direction_difference_nwp_at_best_fit_level", np.dtype("<f4")),
+    ("quality_index_with_forecast", np.dtype("uint8")),
+    ("quality_index_without_forecast", np.dtype("uint8")),
+    ("quality_index_iwwg_value", np.dtype("uint8")),
+    ("tracer_correlation_method", np.dtype("uint8")),
+    ("tracer_type", np.dtype("uint8")),
+    ("height_assignment_method", np.dtype("uint8")),
+    ("orographic_index", np.dtype("uint8")),
+    ("cloud_type", np.dtype("uint8")),
+    ("correlation", np.dtype("uint8")),
+]
+
+# Global attributes accessed by the file handler
+NOMINAL_PRODUCT_TIME = "2025-02-06T13:00:00Z"
+SPATIAL_RESOLUTION = np.array([3.], dtype=np.float32)
+SATELLITE_IDENTIFIER = np.bytes_("MSG3")
+# Dataset attributes accessed by the file handler
+TIME_PERIOD = np.array([15.], dtype=np.float32)
+
+NUM_OBS = 123
+
+FILENAME_INFO = {"platform_id": "MSG3", "region_id": "MSG-N-BS", "start_time": dt.datetime(2025, 2, 6, 13, 0)}
+FILETYPE_INFO = {"file_type": "nc_nwcsaf_geo_hrw"}
+
+
+@pytest.fixture
```

Comment:
```suggestion
@pytest.fixture(scope="module")
```

Unless I misunderstand the fixture, this should make it so it is only created once for all of this module's tests.

Reply:
I agree with the scope, but this will break `tmp_path`, you need to use `tmp_path_factory` instead https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmp-path-factory-fixture

Reply:
Adjusted in https://github.com/pytroll/satpy/pull/3070/commits/ba584f38eac5e115136006bbd1126c7cb04df886